src/main/java/gregtech/common/EventHandlers.java

```diff
@@ -206,9 +213,92 @@ public static void onLivingEquipmentChangeEvent(LivingEquipmentChangeEvent event
         }
     }
 
+    @SuppressWarnings({ "ConstantValue", "deprecation" })
     @SubscribeEvent
-    @SideOnly(Side.CLIENT)
     public static void onPlayerTick(TickEvent.PlayerTickEvent event) {
+        EntityPlayer player = event.player;
+        if (event.phase == TickEvent.Phase.START && !player.world.isRemote) {
+            IAttributeInstance movementSpeed = player.getEntityAttribute(SharedMonsterAttributes.MOVEMENT_SPEED);
+            if (movementSpeed == null) return;
+            AttributeModifier modifier = movementSpeed.getModifier(BlockUtility.WALKING_SPEED_UUID);
+
+            double speedBonus;
+            if (!player.onGround || player.isInWater() || player.isSneaking()) {
+                speedBonus = 0;
+            } else {
+                IBlockState state = player.world.getBlockState(new BlockPos(
+                        player.posX, player.getEntityBoundingBox().minY - 1, player.posZ));
+                speedBonus = BlockUtility.WALKING_SPEED_BONUS.getDouble(state);
+                if (speedBonus == 0 &&
```

Comment:
I assume this section is to deal with the deprecated interface. Can we get a comment to remove this section in 2.9?